path_oidc.go

```diff
@@ -44,6 +58,107 @@ func pathOIDC(b *jwtAuthBackend) []*framework.Path {
 	}
 }
 
+func (b *jwtAuthBackend) pathCallback(ctx context.Context, req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
+	state := b.verifyState(d.Raw["state"].(string))
+	if state == nil {
+		return logical.ErrorResponse("expired or missing OAuth state"), nil
+	}
+
+	roleName := state.rolename
+	role, err := b.role(ctx, req.Storage, roleName)
+	if err != nil {
+		return nil, err
+	}
+	if role == nil {
+		return logical.ErrorResponse("role could not be found"), nil
+	}
+
+	config, err := b.config(ctx, req.Storage)
+	if err != nil {
+		return nil, err
+	}
+	if config == nil {
+		return logical.ErrorResponse("could not load configuration"), nil
+	}
+
+	provider, err := b.getProvider(ctx, config)
+	if err != nil {
+		return nil, errwrap.Wrapf("error getting provider for login operation: {{err}}", err)
+	}
+
+	var oauth2Config = oauth2.Config{
+		ClientID:     config.OIDCClientID,
+		ClientSecret: config.OIDCClientSecret,
+		RedirectURL:  state.redirectURI,
+		Endpoint:     provider.Endpoint(),
+		Scopes:       []string{oidc.ScopeOpenID},
+	}
+
+	oauth2Token, err := oauth2Config.Exchange(ctx, d.Raw["code"].(string))
```

Comment:
What would happen here if the code weren't provided, or were provided but as a different type than as a string?

Reply:
Refactored to use the Get() method which will ensure we'll at least always have a string.